spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/observation/AppDeployerDocumentedObservation.java

```diff
@@ -0,0 +1,353 @@
+	 */
+	DEPLOYER_DEPLOY_OBSERVATION {
+		@Override
+		public Class<? extends Observation.ObservationConvention<? extends Observation.Context>> getDefaultConvention() {
+			return DefaultAppDeployerObservationConvention.class;
+		}
+
+		@Override
+		public String getName() {
+			return "spring.cloud.deployer.deploy";
+		}
+
+		@Override
+		public String getContextualName() {
+			return "deploy";
+		}
+
+		@Override
+		public KeyName[] getLowCardinalityKeyNames() {
+			return Tags.values();
+		}
+
+		@Override
+		public Event[] getEvents() {
+			return Events.values();
+		}
+	},
+
+	/**
+	 * Observation created upon undeploying of an application.
+	 */
+	DEPLOYER_UNDEPLOY_OBSERVATION {
+		@Override
+		public Class<? extends Observation.ObservationConvention<? extends Observation.Context>> getDefaultConvention() {
+			return DefaultAppDeployerObservationConvention.class;
+		}
+
+		@Override
+		public String getName() {
+			return "spring.cloud.deployer.undeploy";
+		}
+
+		@Override
+		public String getContextualName() {
+			return "undeploy";
+		}
+
+		@Override
+		public KeyName[] getLowCardinalityKeyNames() {
+			return Tags.values();
+		}
+
+		@Override
+		public Event[] getEvents() {
+			return Events.values();
+		}
+	},
+
+	/**
+	 * Observation created upon asking for a status of a deployed application.
+	 */
+	DEPLOYER_STATUS_OBSERVATION {
+		@Override
+		public Class<? extends Observation.ObservationConvention<? extends Observation.Context>> getDefaultConvention() {
+			return DefaultAppDeployerObservationConvention.class;
+		}
+
+		@Override
+		public String getName() {
+			return "spring.cloud.deployer.status";
+		}
+
+		@Override
+		public String getContextualName() {
+			return "status";
+		}
+
+		@Override
+		public KeyName[] getLowCardinalityKeyNames() {
+			return Tags.values();
+		}
+
+		@Override
+		public Event[] getEvents() {
+			return Events.values();
+		}
+	},
+
+	/**
+	 * Observation created upon asking for statuses of deployed applications.
+	 */
+	DEPLOYER_STATUSES_OBSERVATION {
+		@Override
+		public Class<? extends Observation.ObservationConvention<? extends Observation.Context>> getDefaultConvention() {
+			return DefaultAppDeployerObservationConvention.class;
+		}
+
+		@Override
+		public String getName() {
+			return "spring.cloud.deployer.statuses";
+		}
+
+		@Override
+		public String getContextualName() {
+			return "statuses";
+		}
+		
+		@Override
+		public KeyName[] getLowCardinalityKeyNames() {
+			return Tags.values();
+		}
+
+
+		@Override
+		public Event[] getEvents() {
+			return Events.values();
+		}
+	},
+
+	/**
+	 * Observation created upon asking for logs of deployed applications.
+	 */
+	DEPLOYER_GET_LOG_OBSERVATION {
+		@Override
+		public Class<? extends Observation.ObservationConvention<? extends Observation.Context>> getDefaultConvention() {
+			return DefaultAppDeployerObservationConvention.class;
+		}
+
+		@Override
+		public String getName() {
+			return "spring.cloud.deployer.getLog";
```

Comment:
Curious as to why the name and contextual name is `getLog` versus `log`.

Reply:
It can be `log` - the method name is `getLog` so I wanted to be consistent.